### ISSUE TITLE
Checkstyle 8.1 SuppressionCommentFilter can no longer suppress RegexpMultiline warnings

### ISSUE THREAD

**Issue by Ricq** [checkstyle/checkstyle#4884]
same as #4841

I'm upgrading from Checkstyle 8.0 to 8.1, at which point my Checkstyle workflow breaks. In 8.1 the SuppressionCommentFilter was changed to be a TreeWalker child, instead of a Checker child. With this change it seems that it is now no longer possible to suppress non-TreeWalker warnings in Java files. Specifically, I have quite some RegexpMultiline checks for our code base, for which some locations are disabled through Checkstyle suppression instructions in Java comments.

As an example, consider the following part of a Java file:
```java
	public static void safeShutdown(@CheckForNull ExecutorService executor,
			Duration gracefulTimeout, Duration interruptTimeout, String executorServiceIdentifier) {
			// ...
			// CHECKSTYLE:OFF:RegexpMultiline
			executor.shutdownNow();
			// CHECKSTYLE:ON:RegexpMultiline
			// ...
	}
```

I'm using the following Checkstyle configuration (this is a very small excerpt, but includes the supression and multiline modules in question.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module
  PUBLIC '-//Puppy Crawl//DTD Check Configuration 1.3//EN'
  'http://www.puppycrawl.com/dtds/configuration_1_3.dtd'>
<module name="Checker">
	<property name="severity" value="warning"/>
	<module name="TreeWalker">
		<module name="SuppressionCommentFilter">
			<property name="offCommentFormat" value="CHECKSTYLE\:OFF\:(\w+)"/>
			<property name="onCommentFormat" value="CHECKSTYLE\:ON\:(\w+)"/>
			<property name="checkFormat" value="$1"/>
		</module>
	</module>
	<module name="RegexpMultiline">
		<property name="format" value="\.shutdownNow\(\);(?!\r?\n.*awaitTermination.*)"/>
		<property name="message" value="Don''t use executor.shutdownNow() without awaitTermination. ExecutorUtil.safeShutdown() is usually your best bet."/>
		<property name="fileExtensions" value="java"/>
	</module>
</module>
```

When running this through Checkstyle 8.1 (from Gradle) I now get:
```
[ant:checkstyle] [WARN] X:\dev\projects\toolkit\common-core\src\main\java\nl\forcare\concurrent\DelegatingExecutorService.java:84: Don't use executor.shutdownNow() without awaitTermination. ExecutorUtil.safeShutdown() is usually your best bet. [RegexpMultiline]
[ant:checkstyle] [WARN] X:\dev\projects\toolkit\common-core\src\main\java\nl\forcare\concurrent\ExecutorUtil.java:89: Don't use executor.shutdownNow() without awaitTermination. ExecutorUtil.safeShutdown() is usually your best bet. [RegexpMultiline]
```

---------------

Is there still a way to suppress the RegexpMultiline warnings through comments in the Java files. I am aware that I could rewrite all these locations (hundreds) such that the suppressions become part of the supressions XML file filter, but I'd rather keep using the comments, as they are nice and local in the code and clearly show at that location that we explicitly disable the rule.

Note: although I do not want to prescribe the solution, it would be useful if there would be some generic form of text-file line suppression filter. I could then run Checkstyle over any type of text file (Javascript, Markdown, etc.) and use a certain form of comment in all these files to disable the generic RegexpSingleline module.

**Comment by rnveach** [comment]
This is similar to https://github.com/checkstyle/checkstyle/issues/4841 .

**Comment by rnveach** [comment]
I am closing this as it is now fixed due to #4841 being fixed.

````
$ cat TestClass.java
public class TestClass {
public static void safeShutdown(@CheckForNull ExecutorService executor,
Duration gracefulTimeout, Duration interruptTimeout, String executorServiceIdentifier) {
// ...
// CHECKSTYLE:OFF:RegexpMultiline
executor.shutdownNow();
// CHECKSTYLE:ON:RegexpMultiline
// ...
}
}

$ cat TestConfig.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module
  PUBLIC '-//Puppy Crawl//DTD Check Configuration 1.3//EN'
  'http://www.puppycrawl.com/dtds/configuration_1_3.dtd'>
<module name="Checker">
<property name="severity" value="warning"/>

<module name="SuppressWithPlainTextCommentFilter">
<property name="offCommentFormat" value="CHECKSTYLE\:OFF\:(\w+)"/>
<property name="onCommentFormat" value="CHECKSTYLE\:ON\:(\w+)"/>
<property name="checkFormat" value="$1"/>
</module>
<module name="RegexpMultiline">
<property name="format" value="\.shutdownNow\(\);(?!\r?\n.*awaitTermination.*)"/>
<property name="message" value="Don''t use executor.shutdownNow() without awaitTermination. ExecutorUtil.safeShutdown() is usually your best bet."/>
<property name="fileExtensions" value="java"/>
</module>
</module>

$ java -jar checkstyle-8.7-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````
